chainio/clients/elcontracts/error.go

```diff
@@ -0,0 +1,50 @@
+package elcontracts
+
+import "fmt"
+
+type Error struct {
+	code        int
+	message     string
+	description string
+	cause       error
+}
+
+func (e Error) Error() string {
+	if e.cause != nil {
+		return fmt.Sprintf("%s(%d) - %s: %s", e.message, e.code, e.description, e.cause.Error())
+	}
+	return fmt.Sprintf("%s(%d) - %s", e.message, e.code, e.description)
+}
+
+func (e Error) Unwrap() error {
+	return e.cause
+}
+
+func CreateErrorForMissingContract(contractName string) Error {
+	errDescription := fmt.Sprintf("%s contract not provided", contractName)
+	return Error{1, "Missing needed contract", errDescription, nil}
+}
+
+func CreateForBindingError(bindingName string, errorCause error) Error {
+	errDescription := fmt.Sprintf("Error happened while calling %s", bindingName)
+	return Error{
+		0,
+		"Binding error",
+		errDescription,
+		errorCause,
```

Comment:
Let's use the full syntax here:

```suggestion
		code: 0,
		message: "Binding error",
		description: errDescription,
		cause: errorCause,
```

Reply:
Done in [Use full syntax when creating errors](https://github.com/Layr-Labs/eigensdk-go/pull/477/commits/b56b255007b0fbb1702b04f4b4af83d6fa1199af)